rs/execution_environment/src/canister_manager/tests.rs

```diff
@@ -8089,3 +8088,159 @@ fn node_metrics_history_ingress_query_fails() {
+            compute_allocation_used: state.total_compute_allocation(),
+        };
+
+        let mut wat = r#"
+        (module
+            (memory $memory "#
+            .to_owned();
+
+        wat.push_str(used_wasm_memory.to_string().as_ref());
+        wat.push_str(
+            r#")
+        )"#,
+        );
+
+        let wasm = wat::parse_str(wat).unwrap();
+
+        let sender = canister_test_id(100).get();
+
+        let initial_settings = CanisterSettings {
+            wasm_memory_limit,
+            wasm_memory_threshold: initial_wasm_memory_threshold,
+            ..Default::default()
+        };
+
+        let canister_id = canister_manager
+            .create_canister(
+                canister_change_origin_from_principal(&sender),
+                subnet_id,
+                *INITIAL_CYCLES,
+                initial_settings,
+                MAX_NUMBER_OF_CANISTERS,
+                &mut state,
+                SMALL_APP_SUBNET_MAX_SIZE,
+                &mut round_limits,
+                ResourceSaturation::default(),
+                &no_op_counter(),
+            )
+            .0
+            .unwrap();
+
+        let res = install_code(
+            &canister_manager,
+            InstallCodeContext {
+                origin: canister_change_origin_from_principal(&sender),
+                canister_id,
+                wasm_source: WasmSource::CanisterModule(CanisterModule::new(wasm)),
+                arg: vec![],
+                compute_allocation: None,
+                memory_allocation: None,
+                mode: CanisterInstallModeV2::Install,
+            },
+            &mut state,
+            &mut round_limits,
+        );
+        assert!(res.1.is_ok(), "{:?}", res.1);
+
+        let c_state = res.2.unwrap();
+
+        assert_eq!(
+            c_state.system_state.task_queue.peek_hook_status(),
+            initial_hook_status
+        );
+
+        state.put_canister_state(c_state);
+
+        let settings = CanisterSettingsBuilder::new()
+            .with_wasm_memory_threshold(updated_wasm_memory_threhsold)
+            .build();
+
+        let canister = state.canister_state_mut(&canister_id).unwrap();
+
+        assert!(canister_manager
+            .update_settings(
+                Time::from_nanos_since_unix_epoch(777),
+                canister_change_origin_from_principal(&sender),
+                settings,
+                canister,
+                &mut round_limits,
+                ResourceSaturation::default(),
+                SMALL_APP_SUBNET_MAX_SIZE,
+            )
+            .is_ok());
+
+        assert_eq!(
+            state
+                .canister_state_mut(&canister_id)
+                .unwrap()
+                .system_state
+                .task_queue
+                .peek_hook_status(),
+            updated_hook_status
+        );
+    })
+}
+
+#[test]
+fn update_wasm_memory_threshold_updates_hook_status_ready_to_not_satisfied() {
+    let overhead = 6553600;
```

Comment:
This overhead represents wasm_memory used while installing the canister and setting `wasm_memory_limit` to a smaller value will fail. Does someone know why this happens?